### PR TITLE
Rule Builder

### PR DESCRIPTION
**Pull request by blainsmith**
This work provides `rule.Build()` in order to help create nftable rules with some guardrails while still allowing the creation of rules of arbitrary order if its partials.

`rule.Build()` requires each rule has a statement verdict such as Drop, Accept, etc.

Additionally, you can specify 0...N matches to make the rule more specific to match the traffic you want.

There is also the following `Match` options if you want to use [named sets](https://wiki.nftables.org/wiki-nftables/index.php/Sets) when you want to supply address and ports.

- `rule.SourceAddressSet()`
- `rule.SourcePortSet()`
- `rule.DestinationAddressSet()`
- `rule.DestinationPortSet()`

### CODE REVIEW

File: pkg/rule/builder.go [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+const (
+	AnyTransport TransportProto = -1
+	TCP          TransportProto = unix.IPPROTO_TCP
+	UDP          TransportProto = unix.IPPROTO_UDP
+)
+
+type Verdict expr.VerdictKind
+
+const (
```
Comment (joewilliams) [inline pkg/rule/builder.go:31]: Please add support for the all the possible verdicts. These don't change often and we might as well support all of them https://github.com/google/nftables/blob/main/expr/verdict.go#L42

File: pkg/rule/builder_test.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package rule_test
+
+import (
+	"net/netip"
+	"testing"
+
+	"github.com/ngrok/firewall_toolkit/pkg/rule"
+)
+
+func TestBuilder(t *testing.T) {
```
Comment (joewilliams) [inline pkg/rule/builder_test.go:10]: Please add negative tests as well as rule output validations similar to the tests in rule_test.go.

**Review by joewilliams — CHANGES_REQUESTED** [review]
Please add helpers for `expressions.LoadCtByKey` and `expressions.CompareCtState`

File: pkg/rule/builder_test.go (hunk 2 above)
Comment (blainsmith) [reply pkg/rule/builder_test.go:10]: Agreed. Now that we are happy with the API I will add more extensive test coverage.

**Review by joewilliams — APPROVED** [review]

**Pull request merged**